src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -18,42 +21,62 @@ export interface DialogProps extends ModalProps {
     /** Dialog title */
     title?: string;
     /** Properties of the accept button */
-    acceptButton: ButtonProps;
+    acceptButton?: ButtonProps;
     /** Properties of the cancel button */
     cancelButton?: ButtonProps;
+    /** the context of the buttons in the footer */
+    variant?: Omit<ButtonContext, 'secondary'>;
+    /** closes the dialog */
+    onClose: () => void;
```

Comment:
I think we discussed this before. `onClose` is only needed when there is a title, am I right? If yes, you can require the `onClose` to be always required only when `title` is present. You can take a look at the example of the `Input`, that if you define `context = 'critical'`, the `errorMessage` is required.